### PR TITLE
Make the default shell_out timeout of 30 seconds configurable

### PR DESCRIPTION
**Pull request by WheresAlice**
### Description

This makes the default timeout of 30 seconds configurable

### Issues Resolved

- #1072 

### Check List

- [ ] New functionality includes tests
- [x] All tests pass
- [ ] RELEASE\_NOTES.md, has been updated if required (not required for bugfixes, required for API changes)
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: lib/ohai/config.rb [hunk 1]
```diff
@@ -40,6 +40,7 @@ class Config
       default :run_all_plugins, false
       # optional plugins are the set of plugins that are marked optional but you wish to run.
       default :optional_plugins, []
+      default :timeout, 30
```
Comment (tas50) [inline lib/ohai/config.rb]: Can you change this to shellout_timeout so it's more clear what it changes?

**Comment by coderanger** [comment]
Is there a specific use case in mind? Are you trying to make it longer? Shorter? Not against the patch, but we should have a goal in mind :)

File: lib/ohai/config.rb (hunk 1 above)
Comment (lamont-granquist) [reply lib/ohai/config.rb]: +1 on using a more specific name

**Comment by WheresAlice** [comment]
I'm thinking specifically around the use-case in #1072 - like the original reporter, we are also seeing chef runs failing on some of our systems with a lot of large network storage mounts.  At the moment we are having to manually patch this code after every upgrade of Chef client in order to make it work on some of our servers.

File: lib/ohai/config.rb (hunk 1 above)
Comment (WheresAlice) [reply lib/ohai/config.rb]: Fair point, will do.

**Comment by cheeseplus** [comment]
I like this one and the use case seems legit - I know we've seen this with other customers and clients

**Comment by WheresAlice** [comment]
The existing tests check that the timeouts are set to 30 seconds, which seems to cover this change.  So I'm not sure I need to add any additional tests.

Should I be adding something to RELEASE_NOTES.md for this?

**Comment by tas50** [comment]
I'll handle the expeditor fixes in another PR

**Pull request merged**